xskillscore/core/resampling.py

```diff
@@ -0,0 +1,211 @@
+import dask
+import numpy as np
+import xarray as xr
+
+CONCAT_KWARGS = {"coords": "minimal", "compat": "override"}
+
+
+def _gen_idx(forecast, dim, iterations, select_dim_items, replace, new_dim):
+    """Generate indices to select from. replace decides whether resampling is with or
+    without replacement."""
```

Comment:
I think it'd be helpful to add a brief descriptor of the inputs here, since there's a lot of them. You don't need to do the full `Parameter` documentation styling, but at least what `select_dim_items` does and `new_dim`.